src/app.ts

```diff
@@ -25,26 +22,31 @@ async function main(): Promise<void> {
     }
   }
 
-  const pageLinks: PageTitleAndLink[] = await fetchLessonUrls(COURSE_URL);
-
-  let i = 1;
-  let promises = [];
-  for (const page of pageLinks) {
-    try {
-      if (i % 5 === 0 || i === pageLinks.length) {
-        await Promise.all(promises.map((p) => p));
-        promises = [];
-      }
-
-      promises.push(downloadPage(`${i}.${page.title}`, page.link));
-      i++;
-    } catch (error) {
-      console.error(error.message);
+  if (!DOWNLOAD_ALL_COURSES && !COURSE_URL) {
```

Comment:
We need to do only one thing out of these two, either download all courses available or download with course url. I can split them in if, else if and else conditions. But I am not sure what you are asking here.

`@shihabmridha - Your Comment was - We can do if (!DOWNLOAD_ALL_COURSES && ! COURSE_URL). Because, I always like to avoid else statement.`